File: aula4/cpp_logic/cadastro_estoque.cpp

#include <iostream>
#include <string> // Necessário importar para usar variáveis do tipo texto em C++

using namespace std;

int main() {
    string nome_do_produto;
    int quantidade_em_estoque;
    double preco_unitario;
    
    cout << "--- Cadastro de Produto no MVP ---" << endl;
    
    cout << "Nome do produto: ";
    // getline captura a string inteira, mesmo se houver espaços no nome do produto
    getline(cin, nome_do_produto);
    
    cout << "Quantidade em estoque: ";
    cin >> quantidade_em_estoque;
    
    cout << "Preco unitario: ";
    cin >> preco_unitario;
    
    // Validação de entrada para proteção da regra de negócio
    if (quantidade_em_estoque < 0) {
        cout << "Erro: A quantidade nao pode ser um valor negativo. Por favor, tente novamente." << endl;
    } else {
        cout << "\n--- Resumo do Produto Cadastrado ---" << endl;
        cout << "Produto: " << nome_do_produto << endl;
        cout << "Quantidade: " << quantidade_em_estoque << endl;
        cout << "Preco: R$ " << preco_unitario << endl;
    }
    
    return 0;
}

File: aula4/python_logic/cadastro_estoque.py
def cadastrar_produto():
    print("--- Cadastro de Produto no MVP ---")
    
    nome_do_produto = input("Nome do produto: ")
    quantidade_em_estoque = int(input("Quantidade em estoque: "))
    preco_unitario = float(input("Preço unitário: "))

    # Regra de negócio: impedindo a corrupção do banco de dados do MVP com estoque negativo.
    if quantidade_em_estoque < 0:
        print("Erro: A quantidade não pode ser um valor negativo. Por favor, tente novamente.")
    else:
        # Se os dados são íntegros, geramos o relatório do produto.
        print("\n--- Resumo do Produto Cadastrado ---")
        print(f"Produto: {nome_do_produto}")
        print(f"Quantidade: {quantidade_em_estoque}")
        print(f"Preço: R$ {preco_unitario:.2f}")

cadastrar_produto()